server/src/main/java/org/apache/iotdb/db/engine/storagegroup/DataRegion.java

```diff
@@ -2332,6 +2332,9 @@ private boolean canSkipDelete(
     }
 
     for (PartialPath device : devicePaths) {
+      if (device.hasWildcard()) {
```

Comment:
check if the time cost of precise matching is tolerable.

Reply:
Basically, the performance of query has higher priority than that of data deletion. From the experiment result, it costs 280ms to execute precise matching in memory for 10000 devices，each having 1 measurement and 1000 points.